### PR TITLE
Update dev docs with WSL instructions

### LINKED ISSUE #226: Add WSL setup directions to README

### PR DESCRIPTION
**Pull request by julesfll**
# Status
Done

However, the bottom section could use revision as well, to make sure deploy instructions are up to date

# What I did
Added WSL instructions for installation

## Issues addressed
Vagrant is no longer necessary since WSL is now supported

Closes #226

### CODE REVIEW

**Comment by julesfll** [comment]
Down the line, it'd be great if @jackliu612 / @Bartleby2718  / @alex-shen1 could make sure the **Site updates** section is up to date

**Comment by alex-shen1** [comment]
BTW @julesfll for future reference, the repo is now set up so that you can write something like `Closes #226` in the original PR description and it should automatically link the issue and close it once the PR is merged (see [this page](https://github.blog/2013-05-14-closing-issues-via-pull-requests/) for details) On the other hand, not sure if manually linking actually results in the issue being closed automatically; we've had some floating issues remain in the board that should've been closed.

**Comment by alex-shen1** [comment]
Technically outside the scope of this PR, the **Site Updates** portion was *incredibly* out of date (it even referenced a command that either doesn't exist anymore or never existed in the first place, yikes) so I went ahead and fixed it.

File: FOR-DEVS.md [hunk 1]
```diff
@@ -68,13 +68,20 @@ If you are part of theCourseForum engineering team, follow the instructions belo
     where `db_dump_file.sql` should be replaced with the latest DB file (e.g. `launch.sql`)
 7. Go to http://localhost:8000 in your browser and make sure everything works!
 
+### Windows setup (recommended)
+1. Install [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+2. Install Docker
+    - If you're using Windows 10 Pro, Enterprise, or Education, then follow [these instructions](https://docs.docker.com/docker-for-windows/install/) to install Docker Desktop, which comes with Docker and Docker Compose. A free copy of Windows 10 Education is available from UVA.
```
Comment (Bartleby2718) [inline FOR-DEVS.md:74]: Maybe add an instruction for how to check the OS?

File: FOR-DEVS.md [hunk 2]
```diff
@@ -68,13 +68,20 @@ If you are part of theCourseForum engineering team, follow the instructions belo
...
+3. Proceed at step 2 in **MacOS and Linux setup**
+
 ### Alternative Setup (Windows and MacOS if above failed)
 0. [Install Vagrant](https://www.vagrantup.com/intro/getting-started/install.html)
 1. `git clone https://github.com/thecourseforum/theCourseForum2.git`
 2. Go into the `theCourseForum2/` folder and then run `vagrant up` to start a VM.
 3. After that's booted, run `vagrant ssh` to run commands inside the VM.
 4. `cd /vagrant`
-5. Start at step 3 in normal setup. You may have to prefix all docker commands with `sudo` (e.g. `sudo docker-compose up`).
+5. Start at step 3 in **MacOS and Linux setup**. You may have to prefix all docker commands with `sudo` (e.g. `sudo docker-compose up`). ----------------------
```
Comment (Bartleby2718) [inline FOR-DEVS.md:84]: What is this line?

File: FOR-DEVS.md [hunk 3]
```diff
@@ -154,11 +161,15 @@ python3 manage.py migrate
...
+5. I think we have to run the `department_fixes` script too? But not sure about that one.
+6. TODO: Consolidate this into one master script that we can run as a cron job on prod.
+
+<sub>\* Yeah, this method really sucks; should probably improve it in the consolidation process</sub>
+
+### Deploying to prod
+Production deployments (i.e. the version of the site users see on thecourseforum.com) should be handled automatically by our CI (continuous integration system). As of the writing of this documentation, Travis is set up to autodeploy every time a build on the `master` branch passes and we're working on moving towards GitHub Actions. However, in the event that something fails, the manual deploy process is outlined [here](https://docs.google.com/document/d/1sPl1v4JrvicrgQluXvG9cR6GWPijC0hr7zAa3h3uE5E/edit#heading=h.9mo53b4db0s8). Just note that the environmental variables list in the doc may not be up-to-date, so remember to set them accordingly.
```
Comment (Bartleby2718) [inline FOR-DEVS.md:174]: @alex-shen1  I think you meant `continuous delivery` here? 
https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment

**Review by Bartleby2718 — CHANGES_REQUESTED** [review]

File: FOR-DEVS.md (hunk 3 above)
Comment (alex-shen1) [reply FOR-DEVS.md:174]: oh shoot lol i conflated them

File: FOR-DEVS.md (hunk 2 above)
Comment (julesfll) [reply FOR-DEVS.md:84]: Oh lol that was a dumb way to remind me to change that line
```suggestion
5. Start at step 3 in **MacOS and Linux setup**. You may have to prefix all docker commands with `sudo` (e.g. `sudo docker-compose up`).
```

File: FOR-DEVS.md (hunk 1 above)
Comment (julesfll) [reply FOR-DEVS.md:74]: Done

**Review by Bartleby2718 — APPROVED** [review]
LGTM, but I think @jackliu612 *might* have  something to add once he actually uses this to enable WSL2.

File: FOR-DEVS.md [hunk 4]
```diff
@@ -154,11 +161,14 @@ python3 manage.py migrate
...
+1. Go into the `/tcf_website/management/commands/semester_data` directory. Edit `fetch_data.py` to run `download_semester()` on the semester(s) you want*. For example, if you wanted Fall 2020, you would add a line saying `download_semester(2020, 'fall')`.
+2. Commit this updated data and push it into the repo; it needs to be on prod to update the DB (you could update locally, but what's the point?). 
+3. On a prod environment (we've been using `heroku run bash --app thecourseforum-dev` because our prod/dev DBs are linked...but we probably won't have it be this way forever), run `python3 manage.py load_semester ${SEMESTER}`, where `${SEMESTER}` is the semester you're adding in the format `YEAR_SEASON` e.g. `2020_FALL`. 
+4. You should be able to run this command as many times as you want, as it accounts for/skips over duplicates and will update sections with new instructors as necessary. Most courses will likely not change much, but this duplicate handling helps us deal with classes whose instructors change after we load the data the first time.
+5. I think we have to run the `department_fixes` script too? But not sure about that one.
```
Comment (jackliu612) [inline FOR-DEVS.md:168]: Doh, I think this won't work with how `department_fixes` is written

**Review by jackliu612 — COMMENTED** [review]
Look good so far, I will try to follow the instructions later today/tomorrow to make sure it at least works for me and I will approve the PR

**Review by jackliu612 — APPROVED** [review]
Worked through the steps and it seems to work for me. I did leave a comment about how `departmnet_fixes` probably does not work in subsequent semester. This is a issue that I can try to address later but work around now would be to load all the semesters all the time.....

**Comment by Bartleby2718** [comment]
@alex-shen1 Maybe we should create a new branch for the additional instructions @jackliu612 isn't sure about?
@jackliu612 Also, does that mean we can take out all Vagrant-related files in the repo? 👀

**Pull request merged**